File: TestHotFix/app/src/main/java/com/example/testhotfix/MyApplication.java
package com.example.testhotfix;

import android.app.Application;
import android.content.Context;

import androidx.multidex.MultiDex;

import com.example.testhotfix.fixmethod.FixUtil;

public class MyApplication extends Application {

    @Override
    protected void attachBaseContext(Context base) {
        super.attachBaseContext(base);
        MultiDex.install(this);
        FixUtil.fix(this, getCacheDir().getAbsolutePath() + "/patch.jar");
    }
}

File: TestHotFix/app/src/main/java/com/example/testhotfix/MyApplication.kt
package com.example.testhotfix

import android.app.Application
import android.content.Context

class MyApplication : Application() {

    override fun attachBaseContext(base: Context?) {
        super.attachBaseContext(base)
        FixUtil.fix(this, cacheDir.absolutePath + "/patch.jar")
    }
}